.github/workflows/aws_publisher.yaml

```diff
@@ -46,6 +50,27 @@ jobs:
         with:
           command: build
           arguments: "-color=false -on-error=abort -var=kafka_ui_release_version=${{ github.event.inputs.KafkaUIReleaseVersion }}"
-          target: kafka-ui-infra/ami/kafka-ui.pkr.hcl
+          target: kafka-ui.pkr.hcl
+          working_directory: ${{ env.WORK_DIR }}
         env:
           PACKER_LOG: 1
+
+      # add fresh AMI to AWS Marketplace
+      - name: Publish Artifact at Marketplace
+        if: ${{ github.event.inputs.PublishOnMarketplace == true }}
+        env:
+          PRODUCT_ID: ${{ secrets.AWS_SERVER_PRODUCT_ID }}
+          RELEASE_VERSION: "${{ github.event.inputs.KafkaUIReleaseVersion }}"
+          RELEASE_NOTES: "https://github.com/provectus/kafka-ui/releases/tag/v${{ github.event.inputs.KafkaUIReleaseVersion }}"
+          MP_ROLE_ARN: ${{ secrets.AWS_MARKETPLACE_AMI_ACCESS_ROLE }} # https://docs.aws.amazon.com/marketplace/latest/userguide/ami-single-ami-products.html#single-ami-marketplace-ami-access
+          AMI_OS_VERSION: "amzn2-ami-kernel-5.10-hvm-*-x86_64-gp2"
```

Comment:
does it have to be x64? We can run on arm as well

Reply:
AFAIK the AMI is platform specific, you may choose to distribute AMIs for more than one platform but that will require another packer build step and probably another publish(there is no mentioning in the documentation how to publish product which runs on both amd64 and arm64).